code/script.js

```diff
@@ -218,108 +248,246 @@ const generateBoard = () => {
           <button class="filled-button small" onclick="guess('${person.name}')">Guess</button>
         </div>
       </div>
-    `
-  })
-}
+    `;
+  });
+};
 
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+  secret =
+    charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+};
+let startTime;
+let elapsedTime = 0;
+let timerInterval;
+
+// A function that resets the timer and then runs the start function
+const reset = () => {
+  clearInterval(timerInterval);
+  print("00:00:00");
+  elapsedTime = 0;
+  start();
+  counter = 1; // resets the counter
+  document.getElementById("counter").innerHTML = `This far you have guessed 0 times`; // resets the innerHTML when the reset is run
+  cheer.pause() // If the user wins the cheer plays and this pauses it when everything resets
+  playElevatorMusic() // this runs the function that has the elevator music
 }
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
+  charactersInPlay = CHARACTERS;
   // What else should happen when we start the game?
+  setTimeout(() => generateBoard(), 500);
+  setSecret();
+  winOrLose.style.display = "";
+  startTimer();
+};
+
+// Gets the time and converts it into strings
```

Comment:
Great that you added a timer function! I can see that the function gets the job done, but it is a bit difficult to follow it when reading it.